backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -149,26 +155,20 @@ private async void OnEmergencyButtonPressedForRobot(object? sender, EmergencyBut
                 return;
             }
 
-            var area = await AreaService.ReadById(robot.CurrentArea!.Id);
-            if (area == null)
-            {
-                _logger.LogError("Could not find area with ID {AreaId}", robot.CurrentArea!.Id);
-                SignalRService.ReportSafeZoneFailureToSignalR(robot, $"Robot {robot.Name} was not correctly localised. Could not find area {robot.CurrentArea.Name}");
-                return;
-            }
-
             try { await MissionScheduling.FreezeMissionRunQueueForRobot(e.RobotId); }
             catch (RobotNotFoundException) { return; }
 
+            var area = await FindRobotArea(robot.Id);
```

Comment:
Currently it is abit unclear from the name that this function also checks what area the robot will be localized in, could the name be more explicit? Something along the lines of FindCurrentRobotAreaOrTheOneItWillBeLocalizedIn, just better formulated?

Reply:
Maybe FindRelevantRobotAreaForSafePositionMission?